src/vm/jithelpers.cpp

```diff
@@ -4968,6 +4968,24 @@ HCIMPL0(void, JIT_ThrowPlatformNotSupportedException)
 }
 HCIMPLEND
 
+/*********************************************************************/
+HCIMPL0(void, JIT_ThrowTypeNotSupportedException)
+{
+    FCALL_CONTRACT;
+
+    /* Make no assumptions about the current machine state */
+    ResetCurrentContext();
+
+    FC_GC_POLL_NOT_NEEDED();    // throws always open up for GC
+
+    HELPER_METHOD_FRAME_BEGIN_ATTRIB_NOPOLL(Frame::FRAME_ATTR_EXCEPTION);    // Set up a frame
+
+    COMPlusThrow(kNotSupportedException, W("NotSupported_Type"));
```

Comment:
This behavior is slightly different from `S.N.Vector<T>` that throws NSE with message "Specified type is not supported".

https://github.com/dotnet/coreclr/pull/15028

Reply:
This is the message the VM typically uses for this exception.  I see no reason it has to match `S`N.Vector<T>`.

Reply:
The only difference between the two is the word "Specified", which might make it more clear that the type itself(`Vector128<T>`) is supported, its only the type of the generic which is not.

I don't have a strong preference either way, however.

Reply:
> I see no reason it has to match S.N.Vector<T>

The only reason is that we may implement `S.N.Vector<T>` via managed code in the future.

Reply:
> The only reason is that we may implement S.N.Vector<T> via managed code in the future.

Even then, we could just manually validate ahead of time (which we probably would need to do anyways, since we have "exploded" rather than purely generic APIs, due to the ISA splits).

Reply:
The other reason would be to match 
```C#
ThrowHelper.ThrowNotSupportedExceptionIfNonNumericType<T>()
```
 which is currently using `SR.Arg_TypeNotSupported`

Reply:
It would seem to make sense to me to match `ThrowHelper.ThrowNotSupportedExceptionIfNonNumericType<T>()`

Reply:
Done